### ISSUE TITLE
Compiler classes are missing standard flags

### ISSUE THREAD

**Issue by white238** [LLNL/spack#33272]
### Summary

The compiler classes are missing several standard flag properties. 

### Rationale

This is causing some hard-coded standard flags in some of the packages which could cause some standard flags to be passed to compilers that do not support that specific flag.  This would cause a build error instead of a more easily understood Spack error message.

### Description

For example, the intel compiler class only has 4:

https://github.com/spack/spack/blob/f24c135383399c5ea4a78744470b66dd729a1601/lib/spack/spack/compilers/intel.py#L68-L100

when Intel@19.0.4 supports the following:

```
-std=<std>
          enable language support for <std>, as described below
            c99   conforms to ISO/IEC 9899:1999 standard for C programs
            c11   conforms to ISO/IEC 9899:2011 standard for C programs
            c++11 enables C++11 support for C++ programs
            c++14 enables C++14 support for C++ programs
            c++17 enables C++17 support for C++ programs
            c++2a enables experimental C++20 support for C++ programs
            c89   conforms to ISO/IEC 9899:1990 standard for C programs
            gnu89 conforms to ISO C90 plus GNU extensions 
            gnu99 conforms to ISO C99 plus GNU extensions 
            gnu++98 conforms to 1998 ISO C++ standard plus GNU extensions 
            gnu++11 conforms to 2011 ISO C++ standard plus GNU extensions
            gnu++14 conforms to 2014 ISO C++ standard plus GNU extensions
```

Examples of some of the hard-coded standard flags:

```
$ grep -R gnu99 *
bridger/package.py:            flags.append("-std=gnu99")
c-blosc/gcc.patch:+    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99")
c-blosc/gcc.patch:-        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99")
cmake/intel-c-gnu11.patch:+  set(CMAKE_C99_EXTENSION_COMPILE_OPTION "${_std}=gnu99")
ebms/package.py:        cflags = "-g -O3 -std=gnu99"
harfbuzz/package.py:                flags.append("-std=gnu99")
json-cwx/package.py:                "-Wwrite-strings -Wno-unused-parameter -std=gnu99",
libflame/package.py:        # -std=gnu99 at least required, old versions of GCC default to -std=c90
libflame/package.py:            flags.append("-std=gnu99")
lua/package.py:            "CC={0} -std=gnu99 {1}".format(spack_cc, self.compiler.cc_pic_flag),
numactl/package.py:        filter_file("-std=gnu99", "-c99", "Makefile.am")
openwsman/package.py:                flags.append("-std=gnu99")
py-grpcio/package.py:        filter_file("-std=gnu99", "", "setup.py")
py-grpcio-tools/package.py:            filter_file("-std=gnu99", "", "setup.py")
rsbench/package.py:        cflags = "-std=gnu99 -O3"
simplemoc/package.py:        cflags = "-std=gnu99"
swfft/package.py:            # remove -Wno-deprecated -std=gnu99
unifyfs/package.py:                flags.append("-std=gnu99")
xsbench/package.py:            cflags = "-std=gnu99"
```

### Additional information

Minor discussion here:

https://github.com/spack/spack/pull/33037

### General information

- [X] I have run `spack --version` and reported the version of Spack
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by alalazo** [comment]
@white238 I'm closing this old issue since it seems outdated. The framework to declare flags is consistent since we moved compilers into the `builtin` repository. The only exception is the NAG compiler - since we must find somebody using it to fill in the data 🙂 

Tl;DR This was likely fixed in v1.0
